### PR TITLE
[actor] Migrate SyncJobsActor to use ActixWrapper

### PR DESCRIPTION
**Pull request by shreyan-gupta**
This PR is mainly a refactor with no functional changes.

TODO: Need to figure out a good way to handle consolidation between ActixFutureSpawner and TestLoopFutureSpawner. For now just using the respective default in handle function and test loop

Follow up of PR https://github.com/near/nearcore/pull/11214

### CODE REVIEW

File: chain/client/src/sync_jobs_actor.rs [hunk 1]
```diff
@@ -1,74 +1,227 @@
...
+    pub fn new(
+        client_sender: ClientSenderForSyncJobs,
+        sync_jobs_sender: SyncJobsSenderForSyncJobs,
+    ) -> Self {
+        Self { client_sender, sync_jobs_sender }
+    }
 
-    #[perf]
-    fn handle(
+    pub fn spawn_actix_actor(self) -> (actix::Addr<ActixWrapper<Self>>, actix::ArbiterHandle) {
```
Comment (shreyan-gupta) [inline chain/client/src/sync_jobs_actor.rs:78]: Now I'm not sure whether the best way to handle creating actix handler would be to have functions like this or to write something generic?

I'm guessing each handler would have it's own specific requirements? Like for example, here we need to set the mailbox capacity which is actix specific.

File: chain/client/src/test_utils/test_loop/sync_jobs_actor.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+) -> LoopEventHandler<SyncJobsActor, SyncJobsSenderForClientMessage> {
+    LoopEventHandler::new_simple(move |msg, sync_jobs_actor: &mut SyncJobsActor| match msg {
+        SyncJobsSenderForClientMessage::_apply_state_parts(msg) => {
+            sync_jobs_actor.handle(msg);
+        }
+        SyncJobsSenderForClientMessage::_block_catch_up(msg) => {
+            sync_jobs_actor.handle(msg);
+        }
+        SyncJobsSenderForClientMessage::_resharding(msg) => {
+            sync_jobs_actor.handle_resharding_request(msg, &future_spawner);
```
Comment (shreyan-gupta) [inline chain/client/src/test_utils/test_loop/sync_jobs_actor.rs:18]: Can not call the generic handle function here as we need to pass the future_spawner. Will tackle this separately.

File: chain/client/src/client_actor.rs [hunk 3]
```diff
@@ -72,21 +71,17 @@ impl ClientActor {
...
-                ctx.set_mailbox_capacity(SyncJobsActor::MAILBOX_CAPACITY);
-                SyncJobsActor {
-                    actions: SyncJobsActions::new(
-                        self_addr_clone.with_auto_span_context().into_multi_sender(),
-                        ctx.address().with_auto_span_context().into_multi_sender(),
-                    ),
-                }
-            },
+
+        let sync_jobs_sender = LateBoundSender::<SyncJobsSenderForSyncJobs>::new();
```
Comment (robin-near) [inline chain/client/src/client_actor.rs:77]: that's an interesting way to remove the need for "start_in_arbiter".

File: chain/client/src/sync_jobs_actor.rs (hunk 1 above)
Comment (robin-near) [reply chain/client/src/sync_jobs_actor.rs:78]: It's currently adhoc, so we can do it adhoc for now. I'm not worried about it as long as it works.

File: chain/client/src/client_actor.rs (hunk 3 above)
Comment (shreyan-gupta) [reply chain/client/src/client_actor.rs:77]: We still do have start_in_arbiter in `spawn_actix_actor` function call

**Pull request merged**